File: src/java/magma/api/error/JavaError.java
package magma.api.error;

import java.io.PrintWriter;
import java.io.StringWriter;

public record JavaError(Exception e) implements Error {
    @Override
    public String display() {
        var writer = new StringWriter();
        this.e.printStackTrace(new PrintWriter(writer));
        return writer.toString();
    }
}

File: src/c/magma/error/JavaError.c
#include <temp.h>
#include <temp.h>
struct JavaError {
	Exception e;
	struct JavaError new(Exception e){
		struct JavaError this;
		this.e = e;
		return this;
	}
	String display(void* _this_){
		struct JavaError this = *(struct JavaError*) this;
		temp = temp;
		this.e.printStackTrace();
		return writer.toString();
	}
};